src/number_format.js

```diff
@@ -231,17 +231,29 @@ class NumberFormat extends React.Component {
     const inputValue = el.value;
     const {formattedValue,value} = this.formatInput(inputValue);
     let cursorPos = el.selectionStart;
+    
+    cursorPos = this.getCursorPosition(inputValue, formattedValue, cursorPos );    
+    
+    el.value = formattedValue;
+    this.setCaretPosition(el, cursorPos);
+
+    /*
+      setting caret position within timeout of 0ms is required for mobile chrome,
+      otherwise browser resets the caret position after we set it
+      We are also setting it without timeout so that in normal browser we don't see the flickering
+     */
+    if (this.timeout) {
+      clearTimeout(this.timeout);
```

Comment:
Is this required because I guess onChangeHandler will be called on next event cycle before which caret position timeout with 0ms will already be executed